pkg/distributors/test/MerkleRedeem.test.ts

```diff
@@ -0,0 +1,273 @@
+    const merkleTree = new MerkleTree(elements);
+    const root = merkleTree.getHexRoot();
+
+    await merkleRedeem.connect(admin).seedAllocations(bn(1), root, claimBalance);
+
+    const proof = merkleTree.getHexProof(elements[0]);
+
+    const result = await merkleRedeem.verifyClaim(lp1.address, 1, claimBalance, proof);
+    expect(result).to.equal(true);
+  });
+
+  it("doesn't allow an allocation to be overwritten", async () => {
+    const claimBalance = bn('9876');
+
+    const elements = [encodeElement(lp1.address, claimBalance)];
+    const merkleTree = new MerkleTree(elements);
+    const root = merkleTree.getHexRoot();
+
+    await merkleRedeem.seedAllocations(1, root, claimBalance);
+
+    // construct tree to attempt to override the allocation
+    const elements2 = [encodeElement(lp1.address, claimBalance), encodeElement(lp2.address, claimBalance)];
+    const merkleTree2 = new MerkleTree(elements2);
+    const root2 = merkleTree2.getHexRoot();
+
+    const errorMsg = 'cannot rewrite merkle root';
+    expect(merkleRedeem.seedAllocations(1, root2, claimBalance.mul(2))).to.be.revertedWith(errorMsg);
+  });
+
+  it('stores multiple allocations', async () => {
+    const claimBalance0 = bn('1000');
+    const claimBalance1 = bn('2000');
+
+    const elements = [encodeElement(lp1.address, claimBalance0), encodeElement(lp2.address, claimBalance1)];
+    const merkleTree = new MerkleTree(elements);
+    const root = merkleTree.getHexRoot();
+
+    await merkleRedeem.seedAllocations(1, root, bn('3000'));
+
+    const proof0 = merkleTree.getHexProof(elements[0]);
+    let result = await merkleRedeem.verifyClaim(lp1.address, 1, claimBalance0, proof0);
+    expect(result).to.equal(true); //"account 0 should have an allocation";
+
+    const proof1 = merkleTree.getHexProof(elements[1]);
+    result = await merkleRedeem.verifyClaim(lp2.address, 1, claimBalance1, proof1);
+    expect(result).to.equal(true); // "account 1 should have an allocation";
+  });
+
+  describe('When a user has an allocation to claim', () => {
+    const claimBalance = bn('1000');
+    let elements: string[];
+    let merkleTree: MerkleTree;
+
+    beforeEach(async () => {
+      elements = [encodeElement(lp2.address, claimBalance)];
+      merkleTree = new MerkleTree(elements);
+      const root = merkleTree.getHexRoot();
+
+      await merkleRedeem.seedAllocations(1, root, claimBalance);
+    });
+
+    it('Allows the user to claimWeek', async () => {
+      const claimedBalance = bn('1000');
+      const merkleProof: BytesLike[] = merkleTree.getHexProof(elements[0]);
+      await merkleRedeem.connect(lp2).claimWeek(lp2.address, 1, claimedBalance, merkleProof, false);
+
+      const result = await rewardToken.balanceOf(lp2.address);
+      expect(result).to.equal(claimedBalance); //"user should have an allocation";
+
+      const isClaimed = await merkleRedeem.claimed(1, lp2.address);
+      expect(isClaimed).to.equal(true); // "claim should be marked as claimed";
```

Comment:
WDYT about extracting this into `it('claimed weeks should be marked as claimed')`? It'd let us remove the small comments explaining what each assertion means, and individual `it`s are extremely fast due to the shared setup.